### DISCUSSION TITLE
Allowing all remotePatterns when using next/image - is it safe?

### DISCUSSION THREAD

**Discussion by mskwon1** [vercel/next.js#71843] [Help]
### Summary

Our company is using `next/image` for performance/optimization purpose, and some of the images' src comes from unknown hosts. So we are curious if allowing ALL hostnames via `image.remotePatterns` is fine.

According to the docs, it says `remotePatterns` exist in order to protect our next js app from malicious users.
> To protect your application from malicious users, configuration is required in order to use external images. This ensures that only external images from your account can be served from the Next.js Image Optimization API. 

So I want to know what kind of threats exist if we allow all hostnames via `remotePatterns` like below
```
  images: {
    remotePatterns: [
      {
        protocol: 'https',
        hostname: '**',
      },
    ],
  },
```

**Comment by b-a-merritt** [comment]
Hello!

I would also like an answer from next.js because they could provide better reasons. I have a hunch about why they suggest not wildcarding any domain. That said, because no one else is answering, I'll take a guess. 

Looking at `/next/dist/shared/lib/image-loader.js` ( version 15.1.7 on npm here: https://www.npmjs.com/package/next?activeTab=code), you can see that next validates local images, so cross-site scripting with anything leading with `/` would be pretty tricky, if not impossible. We do not need to worry about someone doing something along the following, for example: 

> In some dynamic, user-generated content, someone adds an image with a source `src="/logout"`, which, upon loading for every user, would make a `GET` request to the backend and presumably log everyone out. 

Next also protects against invalid URLs (in the same file, line 43). Validating the URL before mounting the component means a bad actor couldn't add a src with a script tag or an additional `onError` handler to the photo. Therefore, the danger must come from a validly requested `src.`

Similarly, most images cannot be problematic because browsers have security features too. When browsers render an `<img />` tag, a GET request is made to the `src`. They read the MIME of the response or try to infer it based on the image's extension. If the MIME type is not matched, the correct rendering library is not used, and the image is discarded. 

It seems to me that the only danger comes from SVGs. SVGs can contain script tags inside of them and event handlers that could be problematic. There might be other things that SVGs can do that might be troublesome that I am not thinking of right now. 

If you aren't allowing user-generated content on your site, then I think you should be fine having the wildcard. However, if you do, perhaps limit it to a range of image extensions (like `.png`, `.jpg`, etc.)?

However, like I said at the beginning, I think it would be great if next added the reasons why to their documentation.
